### PR TITLE
Surface Razor language version

### LINKED ISSUE #12142: No mention of RazorLangVersion on this page even though warnings point to this page about it
I'm building a project that uses .razor files for the purpose of #blazing and I got this warning:

Warning RAZORSDK1005 One or more Razor component files (.razor) were found, but the project is not configured to compile Razor Components. Configure the project by targeting RazorLangVersion 3.0 or newer. For more information, see https://go.microsoft.com/fwlink/?linkid=868374

If you go to https://go.microsoft.com/fwlink/?linkid=868374, it takes you to this doc (https://docs.microsoft.com/en-us/aspnet/core/razor-pages/sdk?view=aspnetcore-2.2), which doesn't mention RazorLangVersion at all.

It would be nice to have a mention of it, and showing where/how to put it in the CSPROJ.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8e930c44-70f6-8820-1451-d1f781ded7a0
* Version Independent ID: 6a25fc79-b9b3-4bbf-4fde-8e44b3d8014e
* Content: [ASP.NET Core Razor SDK](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/sdk?view=aspnetcore-2.2)
* Content Source: [aspnetcore/razor-pages/sdk.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/razor-pages/sdk.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-razorpages**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #12142

### CODE REVIEW

File: aspnetcore/razor-pages/sdk.md [hunk 1]
```diff
@@ -55,6 +55,16 @@ To use the Razor SDK to build class libraries containing Razor views or Razor Pa
 
 ::: moniker-end
 
+## Razor language version
+
+The Razor parser and code generator version is controlled by the `<RazorLangVersion>` MSBuild property. If the app requires a different Razor language version than the version of the shared framework used by the app, set the version in the app's project file:
```
Comment (NTaylorMullen) [inline aspnetcore/razor-pages/sdk.md]: Hmm, today if a user has a mismatched `RazorLangVersion` 99.99% of the time it's because of mismatched SDK + project or incompatible VS bits.

I don't know if I'd suggest people modify the `RazorLangVersion` directly. @pranavkm I believe you added this error message. Is there a better way we can phrase this or is having a user manually set the `RazorLangVersion` appropriate?

Comment (guardrex) [reply aspnetcore/razor-pages/sdk.md]: I could reword it to say that **_in rare cases_** an app requires this. Something like ...

> The Razor parser and code generator version is controlled by the `<RazorLangVersion>` MSBuild property. In rare cases, an app requires a different Razor language version than the version of the shared framework used by the app. The version is set in the app's project file:

Comment (guardrex) [reply aspnetcore/razor-pages/sdk.md]: ... and another thing we could do is move this new section all the way to the end of the topic to minimize its significance to most scenarios.

Comment (NTaylorMullen) [reply aspnetcore/razor-pages/sdk.md]: > I could reword it to say that **_in rare cases_** an app requires this.

That would be beneficial but i'm still not convinced it's ever necessary pending response from @pranavkm. If it is necessary then yes, i'd rephrase to say **rare** 



> and another thing we could do is move this new section all the way to the end of the topic to minimize its significance to most scenarios.

Assuming it's necessary this would be perfect.

Comment (guardrex) [reply aspnetcore/razor-pages/sdk.md]: The problem was the link ... Check out that issue https://github.com/aspnet/AspNetCore.Docs/issues/12142 ... an error link lands here with no coverage.

File: aspnetcore/razor-pages/sdk.md [hunk 2]
```diff
@@ -100,3 +100,13 @@ The Razor SDK defines two primary targets:
 * By default, the Razor SDK doesn't publish reference assemblies that are required to perform runtime compilation. This results in compilation failures when the application model relies on runtime compilation&mdash;for example, the app uses embedded views or changes views after the app is published. Set `CopyRefAssembliesToPublishDirectory` to `true` to continue publishing reference assemblies.
 
 * For a web app, ensure your app is targeting the `Microsoft.NET.Sdk.Web` SDK.
+
+## Razor language version
+
+The Razor parser and code generator version is controlled by the `<RazorLangVersion>` MSBuild property. In rare cases, an app requires a different Razor language version than the version of the shared framework used by the app. The version is set in the app's project file:
```
Comment (pranavkm) [inline aspnetcore/razor-pages/sdk.md]: When targeting the `Microsoft.NET.Sdk.Web` SDK, `RazorLangVersion` is inferred from the the app's target framework version. For projects targeting the `Microsoft.NET.Sdk.Razor` SDK or in the rare cases that the app requires requires a different Razor language version than the inferred value, a version can be configured by setting the property in the application's project file:

**Comment by guardrex** [comment]
@pranavkm How's that look?

**Comment by guardrex** [comment]
@Rick-Anderson Merge plz ... I can't with the broken WIP validation.

**Comment by Eilon** [comment]
Thanks for the doc fix! This would have been exactly the info I needed when I ran into it.
